packages/web/hooks/one-click-trading/use-one-click-trading-params.ts

```diff
@@ -39,90 +45,143 @@ export function getParametersFromOneClickTradingInfo({
  * allowing for easy access and modification within components. If `oneClickTradingInfo` is provided, it uses
  * this to set initial parameters. Otherwise, it fetches the default parameters from the application's API.
  *
- * The hook also provides a mechanism to reset the parameters to their initial values, which
- * can be either the custom parameters passed at initialization or the fetched default parameters.
+ * The hook provides a mechanism to reset the parameters to their initial values and tracks changes
+ * to the parameters when `trackChanges` is true, comparing them against the initial values.
  *
  * This hook is primarily intended for the one click trading settings modal.
  */
-export const useOneClickTradingParams = ({
+export function useOneClickTradingParams({
   oneClickTradingInfo,
   defaultIsOneClickEnabled = false,
 }: {
   oneClickTradingInfo?: OneClickTradingInfo;
   defaultIsOneClickEnabled?: boolean;
-} = {}) => {
-  const {
-    data: defaultTransaction1CTParams,
-    isLoading,
-    isError,
-  } = api.local.oneClickTrading.getParameters.useQuery();
+} = {}) {
+  const { data: defaultParams, isLoading } =
+    api.local.oneClickTrading.getParameters.useQuery();
 
-  const [transaction1CTParams, setTransaction1CTParams] = useState<
+  const [draft, setDraft] = useState<
     OneClickTradingTransactionParams | undefined
-  >(
-    oneClickTradingInfo
-      ? getParametersFromOneClickTradingInfo({
-          oneClickTradingInfo,
-          defaultIsOneClickEnabled,
-        })
-      : undefined
+  >();
+  const [current, setCurrent] = useState<
+    OneClickTradingTransactionParams | undefined
+  >();
+
+  const [changes, setChanges] = useState<OneClickTradingParamsChanges>([]);
+
+  const sessionOrDefault = useMemo(
+    () =>
+      oneClickTradingInfo
+        ? getParametersFromOneClickTradingInfo({
+            oneClickTradingInfo,
+            defaultIsOneClickEnabled,
+          })
+        : defaultParams,
+    [oneClickTradingInfo, defaultIsOneClickEnabled, defaultParams]
+  );
+
+  const sessionOrDefaultWithEnabled = useMemo(
+    () =>
+      sessionOrDefault && {
+        ...sessionOrDefault,
+        isOneClickEnabled: defaultIsOneClickEnabled,
+      },
+    [sessionOrDefault, defaultIsOneClickEnabled]
   );
```

Comment:
I suggest merging this with sessionOrDefault since we can simply avoid using `isOneClickEnabled`

```suggestion
```